syntax = "proto3";

service ScriptService {
  rpc AddMargin(AddMarginRequest) returns (AddMarginResponse);
  rpc PerformOCR(OCRRequest) returns (OCRResponse);
  rpc StartSpeechToLine(StartRequest) returns (StartResponse);
  rpc StopSpeechToLine(StopRequest) returns (StopResponse);
  rpc StartPerformerTracker(StartRequest) returns (StartResponse);
  rpc StopPerformerTracker(StopRequest) returns (StopResponse);
  rpc GetStatuses(StatusRequest) returns (StatusResponse);
}

message AddMarginRequest {
  string file_path = 1;
  string margin_side = 2;
}

message AddMarginResponse {
  string file_path = 1;
}

message OCRRequest {
  string file_path = 1;
}

message OCRResponse {
  string file_path = 1;
}

message StartRequest {}

message StartResponse {
  bool success = 1;
}

message StopRequest {}

message StopResponse {
  bool success = 1;
}

message StatusRequest {}

message StatusResponse {
  string rpc_status = 1;
  string speech_to_line_status = 2;
  string performer_tracker_status = 3;
}
